### PR TITLE
#2875 - Exclude student dependents born after study end date

### PR DESCRIPTION
**Pull request by dheepak-aot**
# Exclude the dependents born after study end date - Part 1

## Solution
- [x] Added a new script task to all parttime-assessment-*.bpmn to parse and transform the student entered dynamic data for student dependents to `calculatedData`
- [x] The script task converts `studentDataDependants`  into ` inputData` performing basic sanitization like null check and then is  validation to exclude dependents born after offering study end date is performed on sanitized input.

image

## Refactor

- [x] Updated all the existing calculations using `studentDataDependants` to use `calculatedDataDependants`.

image

image

## Upcoming PR

- [ ] E2E Tests

### CODE REVIEW

**Review by andrepestana-aot — APPROVED** [review]
Looks good to me! Good job!

File: sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2023-2024.bpmn [hunk 1]
```diff
@@ -808,6 +802,32 @@
       <bpmn:incoming>Flow_0lp9mvw</bpmn:incoming>
       <bpmn:outgoing>Flow_1pbkzeb</bpmn:outgoing>
     </bpmn:intermediateThrowEvent>
+    <bpmn:sequenceFlow id="Flow_0sp9tv0" sourceRef="Activity_1vwijyc" targetRef="Event_1mkiqso" />
+    <bpmn:scriptTask id="Activity_1vwijyc" name="Define Dependents">
+      <bpmn:extensionElements>
+        <zeebe:script expression="=// Consider only dependents who were born on or before the study end date.&#10;inputDataDependents[date(item.dateOfBirth) &#60;= date(offeringStudyEndDate)]" resultVariable="calculatedDataDependants" />
+        <zeebe:ioMapping>
+          <zeebe:input source="=//TODO: Student appeal variable needs to be considered here.&#10;if studentDataDependants = null &#10;then []&#10;else studentDataDependants" target="inputDataDependents" />
```
Comment (andrepestana-aot) [inline sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2023-2024.bpmn:810]: Sorry, after approving I realized this: The script task should be after  "Define PD/PPD status" as we're going to the calculated variables for the appeals in parallel with that one.

Comment (andrewsignori-aot) [reply sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2023-2024.bpmn:810]: Besides that, I loved the idea of using the task and adjusting the inputs but for this one the appeals are supposed to be handled already, right?

**Review by guru-aot — COMMENTED** [review]
Nice work @dheepak-aot

File: sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2023-2024.bpmn (hunk 1 above)
Comment (dheepak-aot) [reply sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2023-2024.bpmn:810]: As we discussed, created this script task in a way that it is expected to be moved inside list of parallel tasks to be done for request a change.

**Review by guru-aot — APPROVED** [review]
LGTM

**Pull request merged**